File: app/schemas/document.py
from typing import Literal

from pydantic import BaseModel


class DocumentMetadataInput(BaseModel):
    doc_type: Literal["paper", "project"]
    title: str
    authors: str | None = None
    publish_year: int | None = None
    keywords: str | None = None
    project_name: str | None = None
    funding_agency: str | None = None
    execution_period: str | None = None
    tech_stack: str | None = None


class DocumentUploadResponse(BaseModel):
    doc_id: str
    collection: str
    chunks_count: int
    metadata: dict


class DocumentSearchResult(BaseModel):
    doc_id: str
    text: str
    metadata: dict
    distance: float | None = None

File: frontend/src/types/document.ts
export type DocType = "paper" | "project";

export interface DocumentMetadataInput {
  doc_type: DocType;
  title: string;
  authors?: string | null;
  publish_year?: number | null;
  keywords?: string | null;
  project_name?: string | null;
  funding_agency?: string | null;
  execution_period?: string | null;
  tech_stack?: string | null;
}

export interface DocumentUploadResponse {
  doc_id: string;
  collection: string;
  chunks_count: number;
  metadata: Record<string, string>;
}

export interface DocumentSearchResult {
  doc_id: string;
  text: string;
  metadata: Record<string, string>;
  distance: number | null;
}

export interface DocumentSearchResponse {
  query: string;
  collection: string;
  results: DocumentSearchResult[];
}
